### ISSUE TITLE
Crash with Docker 2.4 on MacOs Catalina 10.15.7

### ISSUE THREAD

**Issue by cristiani** [testcontainers/testcontainers-scala#133]
After updating to last docker version, 2.40 (released yesterday), my test environment crashes.
[testcontainers-ryuk] WARN org.testcontainers.utility.ResourceReaper - Can not connect to Ryuk at localhost:32770
java.net.ConnectException: Connection refused (Connection refused)
    at java.net.PlainSocketImpl.socketConnect(Native Method)
    at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
    at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
    at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
    at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
    at java.net.Socket.connect(Socket.java:606)
    at java.net.Socket.connect(Socket.java:555)
    at java.net.Socket.<init>(Socket.java:451)
    at java.net.Socket.<init>(Socket.java:228)
    at org.testcontainers.utility.ResourceReaper.lambda$null$1(ResourceReaper.java:138)
    at org.rnorth.ducttape.ratelimits.RateLimiter.doWhenReady(RateLimiter.java:27)
    at org.testcontainers.utility.ResourceReaper.lambda$start$2(ResourceReaper.java:136)
    at java.lang.Thread.run(Thread.java:748)
The only solution was to return to downgrade docker.

**Comment by LMnet** [comment]
Try to explicitly set testcontainers-java to [1.15.0-rc2](https://github.com/testcontainers/testcontainers-java/releases/tag/1.15.0-rc2) version.

**Comment by cristiani** [comment]
I am using scala, and build.sbt I have dependencies only to  testcontainers-scala-scalatest and testcontainers-scala-postgresql

**Comment by LMnet** [comment]
It's ok, just add explicit testcontainers-java dependency with it.

**Comment by bman654** [comment]
I tried this work-around.  I added this dependency:

```
      "org.testcontainers" % "testcontainers"                  % "1.15.0-rc2"  % Test,
```

and I still get this error:

```
[2020-10-01 12:17:52,329] [INFO] [org.testcontainers.dockerclient.DockerClientProviderStrategy] [ScalaTest-run] [] - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
[2020-10-01 12:17:53,125] [INFO] [org.testcontainers.dockerclient.UnixSocketClientProviderStrategy] [ScalaTest-run] [] - Accessing docker with local Unix socket
[2020-10-01 12:17:53,125] [INFO] [org.testcontainers.dockerclient.DockerClientProviderStrategy] [ScalaTest-run] [] - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
[2020-10-01 12:17:53,279] [INFO] [org.testcontainers.DockerClientFactory] [ScalaTest-run] [] - Docker host IP address is localhost
[2020-10-01 12:17:53,327] [INFO] [org.testcontainers.DockerClientFactory] [ScalaTest-run] [] - Connected to docker: 
  Server Version: 19.03.13
  API Version: 1.40
  Operating System: Docker Desktop
  Total Memory: 7966 MB
[2020-10-01 12:17:53,603] [INFO] [org.testcontainers.utility.RegistryAuthLocator] [ScalaTest-run] [] - Credential helper/store (docker-credential-desktop) does not have credentials for index.docker.io
[2020-10-01 12:17:54,147] [WARN] [org.testcontainers.utility.ResourceReaper] [testcontainers-ryuk] [] - Can not connect to Ryuk at localhost:32772
java.net.SocketException: Broken pipe (Write failed)
	at java.net.SocketOutputStream.socketWrite0(Native Method)
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:111)
	at java.net.SocketOutputStream.write(SocketOutputStream.java:143)
	at org.testcontainers.utility.ResourceReaper$FilterRegistry.register(ResourceReaper.java:424)
	at org.testcontainers.utility.ResourceReaper.lambda$null$1(ResourceReaper.java:152)
	at org.rnorth.ducttape.ratelimits.RateLimiter.doWhenReady(RateLimiter.java:27)
	at org.testcontainers.utility.ResourceReaper.lambda$start$2(ResourceReaper.java:136)
	at java.lang.Thread.run(Thread.java:748)
```

Here's the relevant sbt dependencies:

```
$ sbt test:dependencyList|grep contain
[info] com.dimafeng:testcontainers-scala-core_2.12:0.38.1
[info] com.dimafeng:testcontainers-scala-jdbc_2.12:0.38.1
[info] com.dimafeng:testcontainers-scala-postgresql_2.12:0.38.1
[info] com.dimafeng:testcontainers-scala-scalatest_2.12:0.38.1
[info] org.testcontainers:database-commons:1.14.3
[info] org.testcontainers:jdbc:1.14.3
[info] org.testcontainers:postgresql:1.14.3
[info] org.testcontainers:testcontainers:1.15.0-rc2
```

**Comment by bman654** [comment]
ok seems to be Intellij-specific now.  When I ran `sbt test` I got a different error.  Added more dependencies and it now works from `sbt test`.  I ended up adding all of these dependencies:

```
lazy val testcontainersScalaVersion = "0.38.1"
lazy val testcontainersJavaVersion = "1.15.0-rc2"

// ...

      "com.dimafeng"      %% "testcontainers-scala-scalatest"  % testcontainersScalaVersion % Test,
      "com.dimafeng"      %% "testcontainers-scala-postgresql" % testcontainersScalaVersion % Test,
      // these dependencies fix incompatability issue with docker for mac: https://github.com/testcontainers/testcontainers-scala/issues/133
      "org.testcontainers" % "testcontainers"                  % testcontainersJavaVersion  % Test,
      "org.testcontainers" % "database-commons"                % testcontainersJavaVersion  % Test,
      "org.testcontainers" % "postgresql"                      % testcontainersJavaVersion  % Test,
      "org.testcontainers" % "jdbc"                            % testcontainersJavaVersion  % Test,
```

**Comment by LMnet** [comment]
I want to pay attention to that this is a temporary solution. This issue will be fixed when testcontainers-java released non-rc version. And next testcontainers-scala version will be released with the new testcontainers-java version.

**Comment by danieletorelli** [comment]
@LMnet do you have any ETA

> I want to pay attention to that this is a temporary solution. This issue will be fixed when testcontainers-java released non-rc version. And next testcontainers-scala version will be released with the new testcontainers-java version.

@LMnet do you have any ETA for this by any chance?

**Comment by LMnet** [comment]
@danieletorelli no, I don't have it. You can subscribe to testcontainers-java/scala releases and will receive a notification when a new version will be released.

**Comment by barryoneill** [comment]
There's a workaround documented here:
https://github.com/testcontainers/testcontainers-java/issues/3291

Disable "Use gRPC FUSE for file sharing" until the new version is released.

**Comment by andersbohn** [comment]
Thanks guys, using "TestcontainersJavaVersion = "1.15.0-rc2" worked 👍  
only I also had to monkeypatch `com.dimafeng.testcontainers.PostgreSQLContainer` to be able to use a different postgres image (like before, where it worked with just the image name in a string):
```PostgreSQLContainer(dockerImageOverride = Some(DockerImageName.parse(s"kartoza/postgis:11.5-2.8").asCompatibleSubstituteFor("postgres")))```

**Comment by idontusenumbers** [comment]
@andersbohn can you share your code? I'm new to scala and am trying to replicate that workaround for MySQLContainer.

**Comment by andersbohn** [comment]
> @andersbohn can you share your code? I'm new to scala and am trying to replicate that workaround for MySQLContainer.

sry, cannot, only some pieces or if you have concrete questions maybe:
from `build.sbt`:
```
val TestContainersScalaVersion = "0.38.4"
val TestcontainersJavaVersion = "1.15.0-rc2"

"com.dimafeng"  %% "testcontainers-scala"            % TestContainersScalaVersion % "it",
"com.dimafeng"  %% "testcontainers-scala-postgresql" % TestContainersScalaVersion % "it",
// these dependencies fix incompatability issue with docker for mac: https://github.com/testcontainers/testcontainers-scala/issues/133
"org.testcontainers" % "testcontainers"                  % TestcontainersJavaVersion  % "it",
"org.testcontainers" % "database-commons"                % TestcontainersJavaVersion  % "it",
"org.testcontainers" % "postgresql"                      % TestcontainersJavaVersion  % "it",
"org.testcontainers" % "jdbc"                            % TestcontainersJavaVersion  % "it",
```
from the test:
```
class IntegrationSpec
    extends FlatSpec
    [...]
    with ForAllTestContainer {
lazy val pgContainer = PostgreSQLContainer(dockerImageOverride = Some(DockerImageName.parse(s"kartoza/postgis:11.5-2.8").asCompatibleSubstituteFor("postgres")))
override lazy val container: Container = MultipleContainers(pgContainer, chContainer)
```

**Comment by dimafeng** [comment]
0.38.6 was released with 1.15.0
